arm-software/ci/automerge.py

```diff
@@ -79,15 +80,21 @@ def prefix_current_commit_message(git_repo: Git) -> None:
     git_repo.run_cmd(["commit", "--amend", "--message=" + commit_msg])
 
 
-def merge_commit(git_repo: Git, to_branch: str, commit_hash: str, ignored_paths: list[str], dry_run: bool) -> None:
+def merge_commit(git_repo: Git, to_branch: str, commit_hash: str, ignored_paths: list[str], dry_run: bool, verbose: bool) -> None:
     logger.info("Merging commit %s into %s", commit_hash, to_branch)
     git_repo.run_cmd(["switch", to_branch])
+    if verbose:
+        current_head = git_repo.run_cmd(["log", "--no-walk", "HEAD", "--pretty=reference"])
+        logger.debug("Current HEAD of %s is %s", to_branch, current_head)
     git_repo.run_cmd(["merge", commit_hash, "--no-commit", "--no-ff"], check=False)
     restore_changes_to_ignored_files(git_repo, ignored_paths)
     if has_unresolved_conflicts(git_repo):
         logger.info("Merge failed")
         git_repo.run_cmd(["merge", "--abort"])
         raise MergeConflictError(commit_hash)
+    if verbose:
+        commit_content = git_repo.run_cmd(["diff", "--cached"])
+        logger.debug("Changes being commited:\n%s", commit_content)
     git_repo.run_cmd(["commit", "--reuse-message", commit_hash])
     prefix_current_commit_message(git_repo)
     if dry_run:
```

Comment:
I also think it would be useful to see the output of `git push`, because it will show the "before" and "after" value of the remote branch head in each case.

Reply:
Would it make sense to print the stdout/stderr of each git command when `--verbose` is used? Or is the `git push` output enough?

Reply:
I've included the printing of all git command outputs for now. We can remove them later if we decide it's too verbose.

Reply:
I suppose it depends if we run any commands that can produce very large output, like `git diff`.

I see a couple of `git diff` in the script, but only with `--name-only`. That's probably all right. The list of names might get long _occasionally_, but not all the time!